### PR TITLE
Add a case about virt-admin server-update-tls

### PR DESCRIPTION
**Pull request by Yingshun**
This PR adds a case as below:
[server-update-tls] notify server to update TLS related files online

depends on:
https://github.com/avocado-framework/avocado-vt/pull/2855
https://github.com/avocado-framework/avocado-vt/pull/2854

Signed-off-by: Yingshun Cui <yicui@redhat.com>

### CODE REVIEW

**Comment by Yingshun** [comment]
**Test result:**
_modular daemons:_
```
#  avocado run  --vt-type libvirt --vt-machine-type q35 virt_admin.server_update_tls 
JOB ID     : 62671b6554eedb9caba094415a0cfd1cddf5df83
JOB LOG    : /root/avocado/job-results/job-2020-12-09T05.01-62671b6/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virt_admin.server_update_tls: PASS (120.45 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 121.71 s
```

_monolithic libvirt:_
```
 (.libvirt-ci-venv-ci-runtest-AdgmaF) [root@dell-per740-18 ~]# avocado run  --vt-type libvirt --vt-machine-type q35 virt_admin.server_update_tls 
JOB ID     : 7c7b4c44888f1f06b5cc41f9594ee04c326f51c6
JOB LOG    : /root/avocado/job-results/job-2020-12-09T04.57-7c7b4c4/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virt_admin.server_update_tls: PASS (119.75 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 121.00 s
(.libvirt-ci-venv-ci-runtest-AdgmaF) [root@dell-per740-18 ~]# vim sp.sh
```

**Comment by Yingshun** [comment]
@iccaszhulili  Would you please help review this PR? It's a high priority case.

File: libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                if fake_ip in line:
+                    line = line.replace(fake_ip, server_ip)
+                f2.write(line)
+
+        cmd = ("certtool --generate-certificate --load-privkey "
+               "{0}/serverkey.pem --load-ca-certificate {0}/cacert.pem "
+               "--load-ca-privkey {0}/cakey.pem --template {0}/server2.info "
+               "--outfile {0}/servercert.pem".format(cert_saved_dir))
+        servercert_pem = os.path.join(cert_saved_dir, "servercert.pem")
+        process.run(cmd, shell=True, verbose=True)
```
Comment (kylazhang) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py:78]: Do we need to check the return to make sure the cmd execute pass?

File: libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        tls_obj.conn_setup()
+        tls_obj.auto_recover = True
+
+        # Connection should fail because TLS is set incorrectly
+        ret, output = libvirt.connect_libvirtd(uri)
+        if ret:
+            test.fail("Connection should fail but succeed. ret: {}, output: {}"
+                      .format(ret, output))
+        if "authentication failed" not in output:
+            test.fail("Unalbe to find expected error message. output: %s"
```
Comment (kylazhang) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py:118]: typo fix, thanks

File: libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        result = vp.server_update_tls(serv_name, debug=True)
+        libvirt.check_exit_status(result)
+
+        # Re-connect to the server
+        ret, output = libvirt.connect_libvirtd(uri)
+        if not ret:
+            test.fail("Connection fails, ret: {}, output: {}"
+                      .format(ret, output))
+    finally:
+        logging.debug("Recover test environment")
```
Comment (kylazhang) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py:136]: I think logging.info should be better

Comment (Yingshun) [reply libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py]: updated.

File: libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py (hunk 2 above)
Comment (Yingshun) [reply libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py]: Thanks, updated.

File: libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py (hunk 1 above)
Comment (Yingshun) [reply libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py:78]: If the cmd fails, process will raise an exception, and it's ok for now, we don't need to check the output of cmd.

File: libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        libvirt.check_exit_status(result)
+
+        # Re-connect to the server
+        ret, output = libvirt.connect_libvirtd(uri)
+        if not ret:
+            test.fail("Connection fails, ret: {}, output: {}"
+                      .format(ret, output))
+    finally:
+        logging.info("Recover test environment")
+        remove_remote_firewall_port(tls_port, params)
```
Comment (iccaszhulili) [inline libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py:137]: Do we need to clear the tls env finally?

Comment (Yingshun) [reply libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py:137]: No. When auto_recover = True, GC will clear tls object, so we don't need to do that.

**Review by kylazhang — APPROVED** [review]

File: libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py (hunk 4 above)
Comment (iccaszhulili) [reply libvirt/tests/src/virt_admin/management/virt_admin_server_update_tls.py:137]: Okay

**Review by iccaszhulili — APPROVED** [review]

**Pull request merged**